### PR TITLE
Clean up and move to pnpm

### PR DESCRIPTION
**Pull request by laulauland**
To make life easier, package management better, and CI spiffier we're updating the monorepo to use pnpm for package managemenet and scripts. This only affects the TypeScript parts of this repo.

The main changes are:

- `client-library-otel` is now moved to `packages/` directory that will house all library packages (reusable schemas, types, as well as client libraries)
- the legacy `client-library` is deprecated, source code is removed and README updated to point to `client-library-otel`
- cleans out dead code from frontend and api for deprecated features
- the CI now uses pnpm instead of npm and a global biome to lint

### CODE REVIEW

File: packages/client-library-otel/README.md [hunk 1]
```diff
@@ -36,15 +36,15 @@ While developing within the monorepo, you can run a watcher on the src dir to au
...
+pnpm watch
 ```
 
 Then, point your sample app's package.json to the local build:
 
 ```json
 {
   "dependencies": {
-    "@fiberplane/hono-otel": "file:../fpx/client-library-otel"
+    "@fiberplane/hono-otel": "workspace:*"
```
Comment (brettimus) [inline packages/client-library-otel/README.md:47]: this assumes the sample app is in the monorepo yeah?

**Review by brettimus — APPROVED** [review]
very happy for you or sorry that that happened

**Comment by brettimus** [comment]
i modified a few things, nothing major, looks good to merge, my local workflows all still are okay

File: packages/client-library-otel/README.md (hunk 1 above)
Comment (laulauland) [reply packages/client-library-otel/README.md:47]: yes, I should clarify that in the README

**Review by oscarvz — APPROVED** [review]
👍

**Review by mies — APPROVED** [review]

**Review by mies — APPROVED** [review]

**Pull request merged**